### DISCUSSION TITLE
Any way to improve error ?

### DISCUSSION THREAD

**Discussion by ScreamZ** [amannn/next-intl#1150] [Q&A] [answered]
Hello,

Many time i'm switching from a client component to async server component.

Like in that post https://github.com/vercel/next.js/discussions/55027#discussioncomment-9447825 I get 

`TypeError [ERR_INVALID_STATE]: Invalid state: ReadableStream is already closed`

This can be solved by swapping `useTranslations` with `getTranslations`.

While `useTranslation` is supposed to work on both server and client, sometimes it doesn't, especially when playing with suspense things.

Is there any way to improve the log and warn about this? So we can use `getTranslations` when needed?

Thanks

**Comment by amannn** [comment, marked as answer]
Interesting, haven't encountered this one yet. Could be that there's an upstream issue in Next.js / React, I'd prefer to not advocate for `getTranslations` too much since `useTranslations` [has some benefits](https://next-intl-docs.vercel.app/docs/environments/server-client-components#async-or-non-async).

Note that in the linked issue the user seems to use `useTranslations` in an `async` component, which is not supported (async components [can't use hooks](https://next-intl-docs.vercel.app/docs/environments/server-client-components#async-components)). AFAIK The React Hooks ESLint plugin will [warn about this in the future](https://github.com/facebook/react/pull/27045).

**Reply by ScreamZ** [reply]
Can't wait for this eslint plugin.

I understand your point, seems fair. The eslint plugin should do the job. Maybe I just need my mind to check there is no hooks.

Thank you for you kindness and quick answers each time :)

**Reply by nikolap** [reply]
If it helps, we also had this issue happen only when upgrading from 3.9.5 to 3.10.0 or higher. As a result, we decided to stick with 3.9.5 until a fix comes out
